pkg/build/generator/generator.go

```diff
@@ -432,6 +432,17 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	// Retry getting a non-conflicting name 10 times. If, after 10 tries, there
+	// is still a naming conflict, the process will continue until we try and
```

Comment:
I'm trying not to imply that we should return an error there. I've updated the comment to be more clear about saving the LastVersion, ptal?